process/block/metrics.go

```diff
@@ -70,3 +73,42 @@ func getMetricsFromHeader(
 	appStatusHandler.SetUInt64Value(core.MetricTxPoolLoad, numTxWithDst)
 	appStatusHandler.SetUInt64Value(core.MetricNumProcessedTxs, uint64(totalTx))
 }
+
+func saveMetricsForACommittedBlock(
+	appStatusHandler core.AppStatusHandler,
+	isInConsensus bool,
+	currentBlockHash string,
+	highestFinalBlockNonce uint64,
+	headerMetaNonce uint64,
+) {
+	if isInConsensus {
+		appStatusHandler.Increment(core.MetricCountConsensusAcceptedBlocks)
+	}
+	appStatusHandler.SetStringValue(core.MetricCurrentBlockHash, currentBlockHash)
+	appStatusHandler.SetUInt64Value(core.MetricHighestFinalBlockInShard, highestFinalBlockNonce)
+	appStatusHandler.SetStringValue(core.MetricCrossCheckBlockHeight, fmt.Sprintf("meta %d", headerMetaNonce))
+}
+
+func estimateRewardsForMetachain(
+	publicKeys []string,
+	ownPublicKey []byte,
+	appStatusHandler core.AppStatusHandler,
+	numBlockHeaders int,
+) {
+	isInConsensus := false
+
+	for _, publicKey := range publicKeys {
+		if bytes.Equal([]byte(publicKey), ownPublicKey) {
+			isInConsensus = true
+			continue
+		}
+	}
+
+	if !isInConsensus || numBlockHeaders == 0 {
+		return
+	}
+
+	for i := 0; i < numBlockHeaders; i++ {
```

Comment:
instead of interating over all headers, it would be faster if you would add the length of the `numBlockHeaders` to the current value of the metric

Reply:
Maybe it is faster you suggestion. But from there cannot have access to the current value of the metric so i have to increment the metric (this operation does not take a lot of time because numBlockHeaders is not a big number).

Reply:
My solution here would be to add another function to the status handler interface. Instead of just increment, we can have `IncrementWithVal` (because we can't overload..). Indeed, incrementing in a loop is not very elegant

Reply:
Done. Method AddUint64 method was added in status handler interface.